generators/delimited-schema/main.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+import sys
+import os
+import csv
+import time
+from unicode_csv import UnicodeCsvReader
+
+def main():
+    usage = """Origins Delimited Csv Schema Generator.
+
+    Usage:
+        delimited-schema <file_name> [--encoding=<encoding>] [--delimiter=<delimiter>] [--operation=<assert>|<retract>] [--valid_time=<valid_time>]
+
+    Options:
+        -h --help  Show usage info.
+        --encoding=<encoding>  The text encoding type of the file being read.
+        --delimiter=<delimiter>  The character used to separate items in the csv file.
+        --operation=<assert>|<retract>  The origins operation intended for the new facts. (assert|retract)
+        --valid_time=<valid_time>  The valid time for new facts.
+    """ #noqa
+
+    from docopt import docopt
+
+    # Trim 'origins generate' from the line.
+    argv = sys.argv[2:]
+    args = docopt(usage, argv=argv, version='0.1')
+
+   # Defaults
+    encoding = 'utf-8'
+    delimiter = ','
+    operation = 'assert'
+    valid_time = time.strftime("%Y-%m-%d %H:%M:%S")
+
+    # Collect arg values
+    file_name = os.path.join(os.getcwd(), args['<file_name>'])
+    if args['--encoding'] is not None:
+        encoding = args['--encoding']
+    if args['--delimiter'] is not None:
+        delimiter = args['--delimiter']
+    if args['--operation'] is not None:
+        operation = args['--operation']
+    if args['--valid_time'] is not None:
+        valid_time = args['--valid_time']
+
+    # No need to validate input because docopt makes sure all arguments are given
+    generate(file_name, encoding, delimiter, operation, valid_time)
+
+
+# Minimum possible header.
```

Comment:
Remove this comment since it is not useful and technically not true for the header below.